picard/ui/mainwindow.py

```diff
@@ -946,6 +953,14 @@ def analyze(self):
                 return
         return self.tagger.analyze(self.selected_objects)
 
+    def generate_fingerprints(self):
+        if not config.setting['fingerprinting_system']:
+            if self.show_analyze_settings_info():
+                self.show_options("fingerprinting")
+            if not config.setting['fingerprinting_system']:
```

Comment:
Is `config.setting['fingerprinting_system']` modified in between `if` and here ? It looks weird to me.

Reply:
No, that needs to be changed. I think during development the outer if originally had some other condition or'ed in and I removed that. I'll look into it.

Reply:
Actually the code used to be fine, but I think we broke it at some point when `MainWindow.show_options()` no longer used exec (https://github.com/metabrainz/picard/commit/f8eb75596e8090d2f08e9290608dbe7a9f4964d4). The original idea was to allow the user to enable fingerprinting, and Dialog.exec would halt the code execution until the user closed options again. In this case `config.setting['fingerprinting_system']` indeed could change.

Unfortunately this is no longer happening since PICARD-1571 and the code does not have the desired effect. I added PICARD-1699 to address this. I'll submit a fix in a separate PR.

Reply:
Ok, thanks for explanation, a separate PR will do.